# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/ScreenHelperGTK.h
# Commit: 13eb4d64e83f
# Full Hash: 13eb4d64e83fbc470780e4974392b8fc38c8a008
# Author: stransky <stransky@redhat.com>
# Date: 2025-07-08 16:25:02
# Regressor Bug: 1971161
# File Overlap Count: 1
# Description:
#   Bug 1971161 [Wayland] Move system monitor probe to ScreenGetterGtk class r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D256431
# ==============================================================================

diff -r fa99e2e2b957 -r 13eb4d64e83f widget/gtk/ScreenHelperGTK.h
--- a/widget/gtk/ScreenHelperGTK.h	Tue Jul 08 11:37:54 2025 +0000
+++ b/widget/gtk/ScreenHelperGTK.h	Tue Jul 08 11:41:55 2025 +0000
@@ -15,6 +15,8 @@
 
 namespace mozilla::widget {
 
+class ScreenGetterGtk;
+
 class ScreenHelperGTK final : public ScreenManager::Helper {
  public:
   ScreenHelperGTK();
@@ -23,6 +25,13 @@
   static int GetMonitorCount();
   static gint GetGTKMonitorScaleFactor(gint aMonitorNum = 0);
   static RefPtr<widget::Screen> GetScreenForWindow(nsWindow* aWindow);
+
+  static void RequestRefreshScreens();
+
+ private:
+  static GdkWindow* sRootWindow;
+  static StaticRefPtr<ScreenGetterGtk> gLastScreenGetter;
+  static int gLastSerial;
 };
 
 }  // namespace mozilla::widget
